core/src/retransmit_stage.rs

```diff
@@ -279,6 +288,22 @@ fn retransmit(
                 continue;
             }
 
+            match ShredFetchStage::get_slot_index(packet, &mut ShredFetchStats::default()) {
```

Comment:
Should this logic also account for shred hash so that we retransmit duplicates? i.e. if we retransmit the same shred twice, but then we see a duplicate, we probably want to retransmit the duplicate as well.

Maybe the logic should be, retransmit a single version based on hash of a shred slot/index twice, AND only retransmit at most 2 different hashes of the a shred with the same slot/index

Reply:
Shred fetch stage will filter on shred hash, so only unique shreds should come into here. Then this will allow 2 unique shreds.